lib/commands/session.js

```diff
@@ -13,6 +13,7 @@ commands.createSession = async function (caps) {
   this.validateDesiredCaps(caps);
   this.sessionId = UUID.create().hex;
   this.caps = caps;
+  this.rawCaps = _.cloneDeep(caps); // used for reset
```

Comment:
we should probably add a functional test that each driver runs that asserts that after a test session, `this.caps` remains unchanged.

also, we must not forget to import the basedriver functional tests into each other driver so that they can run the same tests (basedriver functional tests should pass for each sub-driver too!)